tests/test_shards.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (C) 2022 Anaconda, Inc
+# Copyright (C) 2023 conda
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+Test sharded repodata.
+"""
+
+from __future__ import annotations
+
+import json
+import random
+from pathlib import Path
+
+import pytest
+from conda.base.context import context, reset_context
+from conda.core.subdir_data import SubdirData
+from conda.gateways.connection.session import get_session
+from conda.models.channel import Channel
+
+from conda_libmamba_solver.shards import (
+    ShardLike,
+    fetch_shards,
+    shard_mentioned_packages,
+)
+
+HERE = Path(__file__).parent
+
+
+def package_names(shard):
+    """
+    All package names mentioned in a shard (should be a single package name)
+    """
+    return set(package["name"] for package in shard["packages"].values()) | set(
+        package["name"] for package in shard["packages.conda"].values()
+    )
```

Comment:
```suggestion
    return set(package["name"] for package in chain(shard["packages"].values(), shard["packages.conda"].values()))
```
Using `itertools.chain`